python/python/res/config/config_content.py

```diff
@@ -232,3 +233,16 @@ def create_path_elm(self, path):
 
     def add_define(self, key, value):
         self._add_define(key, value)
+
+    def keys(self):
+        return self._keys()
+
+
+    def as_dict(self):
+        d = {}
+        for key in self.keys():
+            d[key] = []
+            item = self[key]
+            for node in item:
+                d[key].append( [ x for x in node ])
+        return d
```

Comment:
I don't completely understand this, but is this equivalent to the following comprehension?
```python
return { key : [list(node) for node in self[key]] for key in self.keys() }
```

I'm not saying you should use mine, but isn't `[x for x in node]` equivalent to `node` or `list(node)`?

Then the `for node in item:` could at least be a comprehension over `self[key]`.

Reply:
Ehhh; I think your construction is the same - but it was quite brutal on the eyes.

Reply:
But `d[key].append( [ x for x in node ])`?